src/components/EnhancedTable/EnhancedTable.tsx

```diff
@@ -62,12 +62,13 @@ export function EnhancedTable<TRow>({
     const { orderBy, orderDirection } = sortParams;
 
     return [...data].sort((first, second) => {
-      const a = first[orderBy];
-      const b = second[orderBy];
-      if (_.isNil(a)) {
+      const a = _.get(first, orderBy);
+      const b = _.get(second, orderBy);
+      // Check if value is either {}, [], undefined, null, but NOT 0 or ''.
+      if (_.isNil(a) || (_.isObject(a) && _.isEmpty(a))) {
         return 1;
       }
-      if (_.isNil(b)) {
+      if (_.isNil(b) || (_.isObject(b) && _.isEmpty(b))) {
```

Comment:
Does `naturalSort` know how to sort versions? I.e. 4.1.3.100 being greater than 4.1.3.9. If not, I'm not sure this solution will be viable. I think you'd be better of just allowing a column to pass in a sorter method so it can specify how it wants to sort.

Reply:
Yeah that was actually what I was going to do originally but thought that I should run through this implementation instead, but you're right, I could just do both (and I'll have to double check the software version sorting, that's a good case)